src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -62,27 +66,50 @@ pub struct LoanDataFlowOperator;
 
 pub type LoanDataFlow<'a, 'tcx> = DataFlowContext<'a, 'tcx, LoanDataFlowOperator>;
 
-pub fn check_crate<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>) {
-    for body_owner_def_id in tcx.body_owners() {
-        tcx.borrowck(body_owner_def_id);
+pub fn check_crate<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, save_analysis: bool) -> Option<HashMap<DefId, AnalysisResult<'a, 'tcx>>> {
+    // FIXME: nashenas88 support this through TyCtxt
+    if save_analysis {
```

Comment:
I don't like this at all. The main reason I didn't modify how `tcx.borrowck` works is that I didn't want to tie the output to the non-mir implementation. I could do the `rls-data` transformations early, and have those types supported in `tcx.borrowck`, but I'm also unsure how you guys would feel about an external library defining the output for such a low level api.